### PR TITLE
Add all currently unused rules

### PR DESCRIPTION
**Pull request by lydell**
See #34.

- [x] Add all deprecated rules, turned off, and marked with `(deprecated)`
- [x] Add new rules that will be turned _off_
- [x] Add new rules that will be turned _on_

@keithamus Which option do you prefer?

1. A single commit for all new rules.
2. Two commits for all new rules: One for the ones turned off, one for the ones turned on.
3. One commit per new rule.

Could you provide commit (a) template(s) that I can use?

### CODE REVIEW

**Comment by keithamus** [comment]
@lydell usually for chunks of rules I've been doing them per-version of eslint, see for example https://github.com/keithamus/eslint-config-strict/commit/ceb16f25f1c776e65ac4c9baf5b806ec74a726ba. Whichever way you feel comfortable doing, I'm happy 😄

File: es5.js [hunk 1]
```diff
@@ -208,8 +212,17 @@ module.exports = {
     'no-proto': 2,
     // disallow declaring the same variable more then once
     'no-redeclare': 2,
+    'no-restricted-properties': [
+      2,
+      {
+        'property': '__defineGetter__',
+        'message': 'Please use Object.defineProperty instead.',
+      },
+    ],
```
Comment (lydell) [inline es5.js]: `.__proto__` is taken care of by the no-proto rule.

Comment (keithamus) [reply es5.js]: `__defineSetter__`, `__lookupGetter__`, `__lookupSetter__` would also be a good to add.

Comment (lydell) [reply es5.js]: I've added those now.

**Comment by keithamus** [comment]
This is looking great, happy for me to merge or still have stuff to tidy up?

**Comment by lydell** [comment]
I think we're good to go!

**Comment by lydell** [comment]
... and this is already out of date. ESLint 3.15.0 brings a new rule: [template-tag-spacing](http://eslint.org/docs/rules/template-tag-spacing). http://eslint.org/blog/2017/02/eslint-v3.15.0-released#new-rules

**Comment by lydell** [comment]
I’ve added template-tag-spacing now.

**Comment by keithamus** [comment]
Amazing work @lydell. Super happy to merge this!

File: es5.js [hunk 2]
```diff
@@ -230,6 +255,10 @@ module.exports = {
     'no-useless-call': 2,
     // disallow unnecessary concatenation of literals or template literals
     'no-useless-concat': 2,
+    // disallow unnecessary escape usage
+    'no-useless-escape': 2,
+    // disallow redundant return statements
+    'no-useless-return': 2,
```
Comment (lydell) [inline es5.js:261]: Just noting that since we enabled this one, it is possible to run into eslint/eslint#7626

Comment (keithamus) [reply es5.js:261]: Hmm, could be problematic. However I'd say both rules are useful, and if someone enters this situation they can always disable whichever rule doesn't fit their usecase with `// eslint-disable-line ...`

**Pull request merged**